okio/src/commonMain/kotlin/okio/internal/Buffer.kt

```diff
@@ -745,9 +745,18 @@ internal inline fun Buffer.commonReadHexadecimalUnsignedLong(): Long {
   return value
 }
 
-internal inline fun Buffer.commonReadByteString(): ByteString = ByteString(readByteArray())
+internal inline fun Buffer.commonReadByteString(): ByteString = readByteString(size)
+
+internal inline fun Buffer.commonReadByteString(byteCount: Long): ByteString {
+  require(byteCount >= 0 && byteCount <= Int.MAX_VALUE) { "byteCount: $byteCount" }
+  if (size < byteCount) throw EOFException()
 
-internal inline fun Buffer.commonReadByteString(byteCount: Long) = ByteString(readByteArray(byteCount))
+  if (byteCount >= Segment.SIZE / 2) {
```

Comment:
This should be a constant somewhere. Suggestions on name and/or location?

Reply:
maybe `SEGMENTING_THRESHOLD`?
immediately after `HEX_DIGIT_BYTES` above?